src/kili/adapters/kili_api_gateway/project/types.py

```diff
@@ -0,0 +1,63 @@
+"""Types for the Project-related Kili API gateway functions."""
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+from kili.domain.project import ComplianceTag, InputType
+from kili.domain.types import ListOrTuple
+
+
+@dataclass
+# pylint: disable=too-many-instance-attributes
+class ProjectDataKiliAPIGatewayInput:
+    """Project input data for Kili API Gateway."""
+
+    archived: Optional[bool]
+    author: Optional[str]
+    compliance_tags: Optional[ListOrTuple[ComplianceTag]]
+    consensus_mark: Optional[float]
+    consensus_tot_coverage: Optional[int]
+    description: Optional[str]
+    can_navigate_between_assets: Optional[bool]
+    can_skip_asset: Optional[bool]
+    honeypot_mark: Optional[float]
+    input_type: Optional[InputType]
+    instructions: Optional[str]
+    json_interface: Optional[str]
+    metadata_types: Optional[Dict]
+    min_consensus_size: Optional[int]
+    number_of_assets: Optional[int]
+    rules: Optional[str]
+    number_of_skipped_assets: Optional[int]
+    number_of_remaining_assets: Optional[int]
+    number_of_reviewed_assets: Optional[int]
+    review_coverage: Optional[int]
+    should_relaunch_kpi_computation: Optional[bool]
+    title: Optional[str]
+    use_honeypot: Optional[bool]
+
+    def __post_init__(self) -> None:
+        _check_argument_ranges(
+            consensus_tot_coverage=self.consensus_tot_coverage,
+            min_consensus_size=self.min_consensus_size,
+            review_coverage=self.review_coverage,
+        )
+
+
+def _check_argument_ranges(
```

Comment:
it should rather be done in the presentation or in the use case. It is linked to what the user inputs

Reply:
I find it weird to handle this here when the role of the gateway should just be to launch a call to Kili API, not to have logic. I would put it in the presentation layer. I asked to GPT and also answered that checks should be in the presentation for separation of concern. I would really avoid to add anything unnecessary in the gateway whose role should only be to send a call to Kili and to adapt the result to the SDK.